geonode/base/api/permissions.py

```diff
@@ -153,6 +154,32 @@ def has_object_permission(self, request, view, obj):
         return IsOwnerOrAdmin.has_object_permission(self, request, view, obj)
 
 
+class IsManagerEditOrAdmin(permissions.BasePermission):
+    """
+    Object-level permission to only allow admin and managers to edit a group.
+    """
+    def has_permission(self, request, view):
+        user = request.user
+        if request.method in ['POST', 'DELETE']:
+            return user and (user.is_superuser or user.is_staff)
+
+        return True
+
+    def has_object_permission(self, request, view, obj):
+        # Read permissions are allowed to any request,
+        # so we'll always allow GET, HEAD or OPTIONS requests
+        if request.method in permissions.SAFE_METHODS:
+            return True
+        user = request.user
+        if user and user.is_superuser or user.is_staff:
+            return True
+        if user and isinstance(obj, GroupProfile) and obj.user_is_role(user, "manager"):
+            if request.method == 'PATCH':
```

Comment:
this `IF` can be moved to the previous line to improve readability since is part of the condition to let the permission be `True`